docker/zig.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+set -x
+set -euo pipefail
+
+# shellcheck disable=SC1091
+. lib.sh
+
+main() {
+    install_packages ca-certificates curl xz-utils
+
+    install_zig
+    install_zigbuild
+
+    purge_packages
+    rm "${0}"
+}
+
+install_zig() {
+    local version="0.9.1"
+    local filename="zig-linux-x86_64-${version}.tar.xz"
+    local dst="/opt/zig"
+
+    local td
+    td="$(mktemp -d)"
+
+    pushd "${td}"
+
+    curl --retry 3 -sSfL "https://ziglang.org/download/${version}/${filename}" -O
+    mkdir -p "${dst}"
+    tar --strip-components=1 -xJf "${filename}" --directory "${dst}"
+
+    popd
+
+    rm -rf "${td}"
+}
+
+install_zigbuild() {
+    local version=0.10.3
+
+    local td
+    td="$(mktemp -d)"
+
+    pushd "${td}"
+
+    export RUSTUP_HOME="${td}/rustup"
+    export CARGO_HOME="${td}/cargo"
+
+    curl --retry 3 -sSfL https://sh.rustup.rs -o rustup-init.sh
+    sh rustup-init.sh -y --no-modify-path --profile minimal
+
+    PATH="${CARGO_HOME}/bin:${PATH}" \
+        cargo install cargo-zigbuild \
+        --version "${version}" \
+        --root /usr/local
```

Comment:
I think building is the right approach, or else we have to:
1. Store the host architecture
2. Convert it to the correct binary

They only offer `*-musl` targets, so although that works fine with our `*-linux-gnu` images, it might get annoying, and we have to consider we won't support on `x86_64-unknown-linux-gnu` base images anymore, necessarily.

Reply:
aha! didn't look too closely